--- /home/yuhei/Documents/SpiderOak/Arch/dmenu-keybind-patch/dmenu-4.5/dmenu.c.orig	2012-01-08 21:18:43.000000000 +0900
+++ /home/yuhei/Documents/SpiderOak/Arch/dmenu-keybind-patch/dmenu-4.5/dmenu.c	2013-03-17 18:16:50.326206105 +0900
@@ -247,7 +247,7 @@
 		switch(ksym) {
 		case XK_a: ksym = XK_Home;      break;
 		case XK_b: ksym = XK_Left;      break;
-		case XK_c: ksym = XK_Escape;    break;
+		case XK_g: ksym = XK_Escape;    break;
 		case XK_d: ksym = XK_Delete;    break;
 		case XK_e: ksym = XK_End;       break;
 		case XK_f: ksym = XK_Right;     break;
@@ -255,8 +255,8 @@
 		case XK_i: ksym = XK_Tab;       break;
 		case XK_j: ksym = XK_Return;    break;
 		case XK_m: ksym = XK_Return;    break;
-		case XK_n: ksym = XK_Down;      break;
-		case XK_p: ksym = XK_Up;        break;
+		case XK_p: ksym = XK_Down;      break;
+		case XK_n: ksym = XK_Up;        break;
 
 		case XK_k: /* delete right */
 			text[cursor] = '\0';
